src/turbomind/models/llama/LlamaWeight.cc

```diff
@@ -51,6 +52,17 @@ LlamaWeight<T>::LlamaWeight(size_t     head_num,
         vocab_size_padded_ = (vocab_size_padded_ + tensor_para_size_ - 1) / tensor_para_size_ * tensor_para_size_;
         TM_LOG_WARNING("pad vocab size from %d to %d", vocab_size_, vocab_size_padded_);
     }
+
+    // try to split embedding table along hidden dim
+    if (hidden_units_ % tensor_para_size_ == 0) {
+        embedding_table_size_ = vocab_size_padded_ * hidden_units_ / tensor_para_size_;
+    }
+    else {
+        embedding_table_size_ = vocab_size_padded_ * hidden_units_;
```

Comment:
Is this case expected to work correctly?

Reply:
I mean we just `FT_CHECK` the condition here and eliminate the else branch since it's not going to work when `hidden_units_ % tensor_para_size_ != 0`